### PR TITLE
feat: optimize blockstream processing

### LINKED ISSUE #353: Evaluate "Unparsed" schema approach
As a Block Node developer
I want to leverage an alternative schema to selectively "shallow parse" protobuf off the wire
So that we can reduce the cost of marshalling all of the objects while also improving maintainability by using an alternative schema rather than the low-level `Bytes` API.

While discussing the low-level approach using the `Bytes` API, Jasper thought we might be able to leverage a different schema to handle selectively parsing levels of objects from the top-down rather than parsing by hand.

### Tech Notes
* Add a new `unparsed.proto` file to the `stream` subproject to specify new types
* Establish the relationship between the new types that mirrors the spec outlined in `hedera-protobufs` but relies in the `bytes` keyword in the definitions to stop PBJ and protoc from processing its children
* Gradle's behavior in the `stream` subproject may need to change to generate types based on the local schema

### PR DESCRIPTION
**Pull request by mattp-swirldslabs**
**Description**:
* Leveraging `unparsed.proto` to define `bytes` types at various levels from the top down to avoid marshaling all the objects
* Converted standard PBJ types to `*Unparsed`
* Changed BlockAsDirWriter and BlockAsDirReader to read/write Bytes
* Added `JProfiler` debugging ability to `update-env.sh` and `docker-compose.yml` for troubleshooting
* Fixed unit tests

**Related issue(s)**:

Fixes #353 #356 

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: server/src/main/java/com/hedera/block/server/pbj/PbjBlockStreamServiceProxy.java [hunk 1]
```diff
@@ -97,15 +99,15 @@ public Pipeline<? super Bytes> open(
             return switch (m) {
                 case publishBlockStream -> {
                     notifier.unsubscribeAllExpired();
-                    yield Pipelines.<PublishStreamRequest, PublishStreamResponse>bidiStreaming()
+                    yield Pipelines.<List<BlockItemUnparsed>, PublishStreamResponse>bidiStreaming()
```
Comment (ata-nas) [inline server/src/main/java/com/hedera/block/server/pbj/PbjBlockStreamServiceProxy.java:102]: Why change from `PublishStreamRequrest` to `List<BlockItemUnparsed>`?

File: server/src/main/java/com/hedera/block/server/persistence/storage/write/NoOpBlockWriter.java [hunk 2]
```diff
@@ -32,27 +30,27 @@
...
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public Optional<List<BlockItem>> write(@NonNull List<BlockItem> blockItems) throws IOException {
+    public Optional<List<BlockItemUnparsed>> write(@NonNull List<BlockItemUnparsed> blockItems) throws IOException {
         if (blockItems.getLast().hasBlockProof()) {
-            metricsService.get(BlocksPersisted).increment();
```
Comment (ata-nas) [inline server/src/main/java/com/hedera/block/server/persistence/storage/write/NoOpBlockWriter.java:55]: Why remove the metrics service?

File: server/src/main/java/com/hedera/block/server/pbj/PbjBlockAccessServiceProxy.java [hunk 3]
```diff
@@ -145,13 +145,12 @@ SingleBlockResponse singleBlock(SingleBlockRequest singleBlockRequest) {
...
-        // TODO: Copying bytes to avoid using references passed from Helidon. Investigate if this is necessary.
-        return SingleBlockRequest.PROTOBUF.parse(Bytes.wrap(message.toByteArray()));
+        return SingleBlockRequest.PROTOBUF.parse(message);
     }
 
     @NonNull
     private Bytes createSingleBlockResponse(
-            @NonNull final SingleBlockResponse reply, @NonNull final RequestOptions options) {
-        return SingleBlockResponse.PROTOBUF.toBytes(reply);
+            @NonNull final SingleBlockResponseUnparsed reply, @NonNull final RequestOptions options) {
```
[the remark this answers is not in the document]
Comment (mattp-swirldslabs) [reply server/src/main/java/com/hedera/block/server/pbj/PbjBlockAccessServiceProxy.java:153]: It's mandated by the PBJ Helidon layer signature.

File: server/src/main/java/com/hedera/block/server/persistence/storage/write/NoOpBlockWriter.java (hunk 2 above)
Comment (mattp-swirldslabs) [reply server/src/main/java/com/hedera/block/server/persistence/storage/write/NoOpBlockWriter.java:55]: This is the NoOpBlockWriter. Tallying the number of blocks persisted when we're not actually doing any writing is misleading

File: server/src/main/java/com/hedera/block/server/pbj/PbjBlockStreamServiceProxy.java (hunk 1 above)
Comment (mattp-swirldslabs) [reply server/src/main/java/com/hedera/block/server/pbj/PbjBlockStreamServiceProxy.java:102]: Our main code in `ProducerBlockItemObserver` is only concerned with a list of inbound items. With the `unparsed.proto` approach, you must selectively step down layer by layer to marshal only the objects you need to make decisions. I've centralized those translations here.

File: server/src/main/java/com/hedera/block/server/pbj/PbjBlockAccessServiceProxy.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (mattp-swirldslabs) [reply server/src/main/java/com/hedera/block/server/pbj/PbjBlockAccessServiceProxy.java:153]: I looked at this again. I think you're right. For unary, I don't think it's required. I'll trim the RequestOptions if it's not required.

**Review by AlfredoG87 — APPROVED** [review]
LGTM 💯

File: server/src/main/java/com/hedera/block/server/pbj/PbjBlockAccessServiceProxy.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (mattp-swirldslabs) [reply server/src/main/java/com/hedera/block/server/pbj/PbjBlockAccessServiceProxy.java:152]: Fixed

File: server/docker/update-env.sh [hunk 4]
```diff
@@ -21,8 +21,9 @@ echo "REGISTRY_PREFIX=" >> .env
 echo "BLOCKNODE_STORAGE_ROOT_PATH=/app/storage" >> .env
 
 if [ true = "$is_debug" ]; then
-  # wait for debugger to attach
-  echo "SERVER_OPTS='-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=*:5005'" >> .env
+  # The server will wait for the debugger to attach on port 5005
+  # JProfiler can attach on port 8849
+  echo "SERVER_OPTS='-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=*:5005' -agentpath:/path/to/libjprofilerti.so=port=8849 " >> .env
```
[the remark this answers is not in the document]
Comment (mattp-swirldslabs) [reply server/docker/update-env.sh:26]: I think we can use the path here if we want the profiler to run remotely in the container. It's not affecting the use of JProfiler from the desktop and bound to the container port. I'll leave it for now and we can adjust it later.

**Review by AlfredoG87 — APPROVED** [review]
LGTM 👍

**Pull request merged**